protocol/auditlog.go

```diff
@@ -0,0 +1,202 @@
+// This module implements a CONIKS audit log that a CONIKS auditor
+// maintains.
+// An audit log is a mirror of many CONIKS key directories' STR history,
+// allowing CONIKS clients to audit the CONIKS directories.
+
+package protocol
+
+import (
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+)
+
+type directoryHistory struct {
+	name      string
```

Comment:
This used to be `dirName` for directory domain name (i.e. its address), which is why we assign it `addr` below. My plan right now is to use this field when auditors query the directories they log every epoch.